### PR TITLE
Allow absolute path in generates section.

### PR DESCRIPTION
**Pull request by smyrman**
Fixes issue #47 and adds a test for it. Absolute paths should also be
allowed in sources after this change, but this has not been specifically
tested.

### CODE REVIEW

File: testdata/generates/Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    - "../rel.txt"
+
+sub/src.txt:
+  desc: generate source file
+  cmds:
+    - mkdir -p sub
+    - echo "hello world" > sub/src.txt
+    # Sleep to avoid time-stamp issue. Some filesystems, E.g. HFS+ for Mac OS,
+    # only support one second precision on file timestamps.
+    - sleep 1
```
Comment (smyrman) [inline testdata/generates/Taskfile.yml]: I will remove this sleep as part of a PR for #49.

**Comment by smyrman** [comment]
Updated to add .gitignore file.

**Comment by andreynering** [comment]
Hi @smyrman 

Both PR's are fine. Thank you for them.

I made a simple comment on #50. You can merge both when you're confident with the code.

**Comment by smyrman** [comment]
Thanks @andreynering. I saw the comment:-)

**Pull request merged**